File: chapter-08-record-pojo/code-java/src/main/java/com/example/demo/dto/BookSummaryDto.java
package com.example.demo.dto;

/**
 * AUTHOR와 BOOK을 JOIN한 결과를 담는 커스텀 DTO.
 * Java 16+ record 문법으로 간결하게 정의합니다.
 */
public record BookSummaryDto(Integer id, String title, String authorLastName) {
}

File: chapter-08-record-pojo/code-kotlin/src/main/kotlin/com/example/demo/dto/BookSummaryDto.kt
package com.example.demo.dto

/**
 * AUTHOR와 BOOK을 JOIN한 결과를 담는 커스텀 DTO.
 * Kotlin data class로 equals/hashCode/toString 자동 생성됩니다.
 */
data class BookSummaryDto(
    val id: Int?,
    val title: String?,
    val authorLastName: String?
)
